src/main/java/app/AppBuilder.java

```diff
@@ -1,52 +1,133 @@
+        LogInPanel logInPanel = new LogInPanel();
+        SignUpPanel signUpPanel = new SignUpPanel();
+
+        // Register panels with ViewManager
+        ViewManager.Instance().registerComponent(logInPanel);
+        ViewManager.Instance().registerComponent(signUpPanel);
 
-        JPanel tradePanel = (JPanel) tradeSimulationFrame.getContentPane().getComponent(0);
-        cardPanel.add(tradePanel, "TradeSimulation");
+        // Add panels to the card layout
+        cardPanel.add(logInPanel, "LogInPanel");
+        cardPanel.add(signUpPanel, "SignUpPanel");
 
         return this;
     }
 
     /**
-     * Build the application
+     * Add the dashboard panel to the application
      *
-     * @return the application
+     * @return the builder
+     */
+    public AppBuilder addDashboardPanel(String username, double cash, double position) {
+        DashboardPanel dashboardPanel = new DashboardPanel(username, cash, position);
+
+        // Register the dashboard panel with ViewManager
+        ViewManager.Instance().registerComponent(dashboardPanel);
+
+        // Add the dashboard panel to the card layout
+        cardPanel.add(dashboardPanel, "DashboardPanel");
+
+        return this;
+    }
+
+    /**
+     * Add the trade simulation panel to the application
+     *
+     * @return the builder
+     */
+    public AppBuilder addTradeSimulationPanel() {
+        TradeSimulationPanel tradeSimulationPanel = new TradeSimulationPanel();
+
+        // Register the trade simulation panel with ViewManager
+        ViewManager.Instance().registerComponent(tradeSimulationPanel);
+
+        // Add the trade simulation panel to the card layout
+        cardPanel.add(tradeSimulationPanel, "TradeSimulationPanel");
+
+        return this;
+    }
+
+    public AppBuilder addDialogComponent() {
+        DialogComponent dialogComponent = new DialogComponent();
+        ViewManager.Instance().registerComponent(dialogComponent);
+        return this;
+    }
+
+    /**
+     * Build the application frame, initialize controllers, interactors, DAOs, and presenters, and register them in ServiceLocator.
+     *
+     * @return the application frame
      */
     public JFrame build() {
-        final JFrame application = new JFrame("Application");
+        JFrame application = new JFrame("Application");
         application.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
         application.setSize(1000, 800);
         application.add(cardPanel);
-        return application;
-    }
 
-    /**
-     * Show the trade simulation
-     */
-    public void showTradeSimulation() {
-        cardLayout.show(cardPanel, "TradeSimulation");
+        //Step 0: Initialize and register InMemoryStockDataAccessObject (DAO)
+        InMemoryStockDataAccessObject stockDataAccessObject = new InMemoryStockDataAccessObject();
+        ServiceManager.registerService(InMemoryStockDataAccessObject.class, stockDataAccessObject);
```

Comment:
I kind of mentioned this idea previously. The current implementation of `AppBuilder` hardcodes the app construction in the source code. A more ideal way is to describe which concrete class injected to which interface in a configuration and build the classes with reflection at runtime.

For the `ServiceManager`, we can create a configuration file (xml is standard for Java, but json may also work well) describing the key-value pairs of <Class>: <Instantiation>. During the initialization of singleton `ServiceManager`, it reads the configuration file and creates the corresponding objects with given class names with reflection. In this way, we can manage the system components in one location (instead of having them scattered in `AppBuilder`), and changes in the concrete classes we use for the system (InMemoryDataAccess or DatabaseDataAccess) only requires changing the configuration.

This feature is optional and can be done in the future. Though I suspect ChatGPT can probably implement it, if you do not have enough time we can mark it as a future refactoring to implement.

Reply:
I'll add a todo for that.